DeepSpeech.py

```diff
@@ -396,29 +395,89 @@ def log_grads_and_vars(grads_and_vars):
+
+'''
+load+list = [put all vars here first]
+init_list=[]
+
+if use_cudnn:
+     move all adam vars to init_list
+if drop_souce_layers>0:
+     move all drop layers to init_list
+if init:
+   override! initialize all vars
+'''
+
+    if FLAGS.load == 'init':
+        session.run(tfv1.global_variables_initializer())
+        return
+    
+    ckpt = tfv1.train.load_checkpoint(checkpoint_filename)
+    load_vars = set(tfv1.global_variables())
+    init_vars = set()
+
+    if use_cudnn:
+        # Load compatible variables from checkpoint    
+        for v in load_vars:
+            try:
+                ckpt.get_tensor(v.op.name)
+            except tf.errors.NotFoundError:
+                load_vars -= v
+                init_vars.add(v)
+
+        # Check that the only missing variables (i.e. those to be initialised)
+        # are the Adam moment tensors
+        if any('Adam' not in v.op.name for v in init_vars):
+            log_error('Tried to load a CuDNN RNN checkpoint but there were '
+                      'more missing variables than just the Adam moment '
+                      'tensors.')
+            sys.exit(1)
+
+    if drop_source_layers>0:
+        '''
+        The transfer learning approach here need us to supply the layers which we
+        want to exclude from the source model.
+        Say we want to exclude all layers except for the first one, we can use this:
+        drop_source_layers=['2', '3', 'lstm', '5', '6']
+        If we want to use all layers from the source model except the last one, we use this:
+        drop_source_layers=['6']
+        '''
+        # FIXME -- if someone tries to drop 6+ layers, this will return an empty list
+        dropped_layers = ['2', '3', 'lstm', '5', '6'][-int(drop_source_layers):]
+
+        # Initialize all variables needed for DS, but not loaded from ckpt
+        for v in load_vars:
+            if any(layer in v.op.name for layer in dropped_layers):
+                init_vars.add(v)
+                load_vars -= v
+                    
+    init_op = tfv1.variables_initializer(init_vars)
+    for v in load_vars:
+        v.load(ckpt.get_tensor(v.op.name), session=session)
+    session.run(init_op)
+
+def try_model
+    
+def find_model(session, load_flag):
+    if load_flag in ['auto', 'last']:
+        checkpoint_path = find_model(session, checkpoint_saver, 'checkpoint', 'most recent')
+    elif load_flag in ['auto', 'best']:
+        checkpoint_path = find_model(session, best_dev_saver, 'best_dev_checkpoint', 'best validation')       
+    elif FLAGS.load in ['auto', 'init']:
+        log_info('Initializing variables...')
+        session.run(initializer)
+    else:
+        log_error('Unable to load %s model from specified checkpoint dir'
+                  ' - consider using load option "auto" or "init".' % FLAGS.load)
         sys.exit(1)
 
+    checkpoint = tf.train.get_checkpoint_state(FLAGS.checkpoint_dir, checkpoint_filename)
+    if not checkpoint:
```

Comment:
`find_model` return value does not seems to be checked, but we should `return None` instead of `return False` right?